acorn/src/state.js

```diff
@@ -132,7 +135,10 @@ export class Parser {
 
   static extend(...plugins) {
     let cls = this
-    for (let i = 0; i < plugins.length; i++) cls = plugins[i](cls)
+    for (let i = 0; i < plugins.length; i++) {
+      cls.plugins.push(plugins[i].name)
```

Comment:
Does this work? `cls` is the constructor here, but it looks like `plugins` is defined as an instance property.

Reply:
Yep, I made a mistake.